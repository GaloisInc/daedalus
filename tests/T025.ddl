-- Test maps

def Main = { @k = UInt8
           ; @v = UInt8
           ; r = Insert k v empty
           ; vr = Lookup 'x' r
           }
