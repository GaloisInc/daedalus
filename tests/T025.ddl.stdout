module T025
 
--- Imports:
 
--- Type defs:
type T025.Main = { r: Map (uint 8) (uint 8)
                 ; vr: uint 8
                 }
 
--- Rules:
 
T025.Main : Grammar T025.Main =
  do (k : uint 8) <- Match UInt8
     (v : uint 8) <- Match UInt8
     (r : Map (uint 8) (uint 8)) <- Insert k v empty
     (vr : uint 8) <- Lookup 'x' r
     pure {r = r,
           vr = vr}
 
T025._Main : Grammar {} =
  do (k : uint 8) <- Match UInt8
     (v : uint 8) <- Match UInt8
     (r : Map (uint 8) (uint 8)) <- Insert k v empty
     @Lookup 'x' r
